skll/learner.py

```diff
@@ -397,33 +403,78 @@ def _train_and_score(learner,
     return train_score, test_score
 
 
-def _predict_binary(self, X):
+def _get_acceptable_regression_metrics():
+    """
+    Return the set of metrics that are acceptable for regression.
     """
-    A helper function to allow us to use ``GridSearchCV`` with objective
-    functions like Kendall's tau for binary classification problems (where the
-    probability of the true class is used as the input to the objective
-    function).
 
-    This only works if we've also taken the step of storing the old predict
-    function for ``self`` as ``predict_normal``. It's kind of a hack, but it saves
-    us from having to override ``GridSearchCV`` to change one little line.
+    # it's fairly straightforward for regression since
+    # we do not have to check the labels
+    acceptable_metrics = (_REGRESSION_ONLY_METRICS |
+                          _UNWEIGHTED_KAPPA_METRICS |
+                          _WEIGHTED_KAPPA_METRICS |
+                          _CORRELATION_METRICS)
+    return acceptable_metrics
+
+
+def _get_acceptable_classification_metrics(label_array):
+    """
+    Return the set of metrics that are acceptable given the
+    the unique set of labels that we are classifying.
 
     Parameters
     ----------
-    X : array-like
-        A set of examples to predict values for.
+    label_array : numpy.ndarray
+        A sorted numpy array containing the unique labels
+        that we are trying to predict. Optional for regressors
+        but required for classifiers.
 
     Returns
     -------
-    res : array-like
-        The prediction results.
+    acceptable_metrics : set
+        A set of metric names that are acceptable
+        for the given classification scenario.
     """
 
-    if self.coef_.shape[0] == 1:
-        res = self.predict_proba(X)[:, 1]
-    else:
-        res = self.predict_normal(X)
-    return res
+    # this is a classifier so the acceptable objective
+    # functions definitely include those metrics that
+    # are specifically for classification and also
+    # the unweighted kappa metrics
+    acceptable_metrics = _CLASSIFICATION_ONLY_METRICS | _UNWEIGHTED_KAPPA_METRICS
+
+    # now let us consider which other metrics may also
+    # be acceptable depending one whether the labels
+    # are strings or (contiguous) integers/floats
+    label_type = label_array.dtype.type
+
+    # CASE 1: labels are strings, then no other metrics
+    # are acceptable
+    if issubclass(label_type, (np.object_, str)):
+        pass
+
+    # CASE 2: labels are integers or floats; the way
+    # it works in SKLL, it's guaranteed that
+    # class indices will be sorted in the same order
+    # as the class labels therefore, ranking metrics
+    # such as various correlations should work fine.
+    elif issubclass(label_type, (int, np.int64, float, np.float64)):
```

Comment:
Hmm, I guess technically it could be. Are there really 32-bit machines out there in the wild that people might use to run machine learning experiments? I guess we never know. I'll include these too.